src/MQTT433gateway.cpp

```diff
@@ -27,251 +27,206 @@
 }
 
-void reconnect() {
-  beatLED.on();
-  // Loop until we're reconnected
-  while (!mqtt.connected()) {
-    Debug(F("Attempting MQTT connection..."));
-    // Attempt to connect
-    if (mqtt.connect(mainTopic.c_str(), mqttUser, mqttPassword,
-                     mainTopic.c_str(), 0, true, "offline")) {
-      DebugLn(F("connected"));
-      mqtt.publish(mainTopic.c_str(), "online", true);
-      mqtt.subscribe((mainTopic + F("/set/+")).c_str());
-      mqtt.subscribe((mainTopic + F("/ota/+")).c_str());
-      mqtt.subscribe((mainTopic + F("/send/+")).c_str());
-      mqtt.subscribe((globalTopic + F("/send/+")).c_str());
-    } else {
-      Debug(F("failed, rc="));
-      Debug(mqtt.state());
-      DebugLn(F(" try again in 5 seconds"));
-      // Wait 5 seconds before retrying
-      beatLED.off();
-      delay(500);
-      beatLED.on();
-      delay(4500);
-    }
+void setupWifi() {
+  beatLED.flash(500);
+  WiFiManager wifiManager;
+  wifiManager.setConfigPortalTimeout(180);
+  wifiManager.setAPCallback([](WiFiManager *) {
+    Logger.info.println(F("Start wifimanager config portal."));
+    beatLED.flash(100);
+  });
+
+  // Restart after we had the portal running
+  wifiManager.setSaveConfigCallback([]() {
+    Logger.info.println(F("Wifi config changed. Restart."));
+    delay(100);
+    ESP.restart();
+  });
+
+  if (!wifiManager.autoConnect(settings.deviceName.c_str(),
+                               settings.configPassword.c_str())) {
+    Logger.warning.println(F("Try connecting again after reboot"));
+    ESP.restart();
   }
   beatLED.off();
 }
 
 void setup() {
   Serial.begin(115200, SERIAL_8N1, SERIAL_TX_ONLY);
+  Logger.addHandler(Logger.DEBUG, Serial);
+  if (!SPIFFS.begin()) {
+    Logger.error.println(F("Initializing of SPIFFS failed!"));
+  }
+
+  settings.registerChangeHandler(MQTT, reconnectMqtt);
+
+  settings.registerChangeHandler(RF_ECHO, [](const Settings &s) {
+    if (rf) rf->setEchoEnabled(s.rfEchoMessages);
+  });
+
+  settings.registerChangeHandler(RF_PROTOCOL, [](const Settings &s) {
+    if (rf) rf->filterProtocols(s.rfProtocols);
+  });
+
+  settings.registerChangeHandler(WEB_CONFIG, [](const Settings &s) {
+    if (!webServer) {
+      setupWebServer(s);
+    } else {
+      webServer->updateSettings(s);
+    }
+  });
+
+  settings.registerChangeHandler(SYSLOG, [](const Settings &s) {
+    if (syslogLog) {
+      Logger.removeHandler(*syslogLog);
+      delete syslogLog;
+    }
+    if (s.syslogLevel.length() != 0 && s.syslogHost.length() != 0 &&
+        s.syslogPort != 0) {
+      syslogLog = new SyslogLogTarget();
+      Logger.addHandler(Logger.stringToLevel(s.syslogLevel), *syslogLog);
+      syslogLog->begin(s.deviceName, s.syslogHost, s.syslogPort);
+    }
+  });
+
+  settings.registerChangeHandler(RF_CONFIG, setupRf);
+
+  settings.registerChangeHandler(LOGGING, [](const Settings &s) {
+    Logger.addHandler(Logger.stringToLevel(settings.serialLogLevel), Serial);
```

Comment:
The Serial logger is not removed, if set to None.

Reply:
Fixed in the next push